src/plugins_types/hex_string.c

```diff
@@ -93,6 +93,7 @@ const struct lyplg_type_record plugins_hex_string[] = {
         .plugin.validate = NULL,
         .plugin.compare = lyplg_type_compare_simple,
         .plugin.print = lyplg_type_print_simple,
+        .plugin.hash = lyplg_type_hash_simple,
```

Comment:
Well, it says `Also, type plugin API adjusted.` and since the hash callbacks just return the canonical value for now (expect for the IP address) I thought it makes sense since it is directly related.

Reply:
yeah, but the hash callbacks themselves are present in another commit. The point is that some of the commits seem as just a rename something or change a particular function and its calls, and such commits are quite easy to review (especially in such huge commits as these), but when you combine several changes together, it's going to be a mess for review. I know, that sometimes (well, often) the changes are somehow connected. But in this case with the hashes, it seems to me that everything related to hashes could be in a single commit, which would simplify the other commits.

Reply:
To be clear - I'm not saying to spend more hours on this PR just to correctly split the commits, just take it more into consideration with the future PRs, this wasn't a nice PR.